### PR TITLE
End to end tests for wormhole integration

### PR DESCRIPTION
**Pull request by mdulin2**
* Standalone with a single endpoint and multiple endpoints
* Standalone while using the relayer
* Upgrade tests to ensure invariant checking and migration works as expected

### CODE REVIEW

**Comment by mdulin2** [comment]
Mostly creating new tests. The only exception is modifying the `migrate()` function to be `virtual` in order to be able to override it during tests for upgrading.

File: test/Upgrades.t.sol [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+            address(relayer)
+        ); 
+        managerChain1.upgradeEndpoint(address(wormholeEndpointChain1), address(wormholeEndpointChain1Implementation)); 
+
+        basicFunctionality();
+    }
+
+
+    // Confirm that we can handle multiple upgrades as a manager
+    function testDoubleUpgradeManager() public {
```
Comment (djb15) [inline test/Upgrades.t.sol:264]: Nit-pick, but can all the test be named like `test_camelCase`

**Comment by djb15** [comment]
Also I guess you need to get all the tests running before we can merge this in 😄

File: test/Upgrades.t.sol (hunk 1 above)
Comment (mdulin2) [reply test/Upgrades.t.sol:264]: The linter lied to me. Can do!

**Comment by mdulin2** [comment]
> Also I guess you need to get all the tests running before we can merge this in 😄

I have the frontrunning of `initialize()` as a test case. I'll just comment it out for the time being.

**Review by djb15 — APPROVED** [review]

**Review by djb15 — APPROVED** [review]

**Review by RahulMaganti47 — APPROVED** [review]

**Pull request merged**